opm/simulators/utils/readDeck.cpp

```diff
@@ -402,15 +403,28 @@ void Opm::ensureOutputDirExists(const std::string& cmdline_output_dir)
     }
 }
 
+void Opm::prepareResultOutputDirectory(const std::string&           baseName,
+                                       const std::filesystem::path& outputDir)  
+{   
+    //Loop over all files in the output directory and subdirectories and delete them if their name is baseName + a correct extension
+    std::regex r(baseName + R"(.(F?(DBG|E?GRID|INIT|PRT|RFT|SMSPEC|UNSMRY|UNRST)|([ABCFGHSTUXYZ]\d{4})|(INFOSTEP|INFOITER|OPMRST)))");
```

Comment:
Tiny nit: The full stop character (`.`) should be escaped here&ndash;i.e., the start of the regular expression should be written as `baseName + R"(\.(F?`.  Otherwise it will match any character and that's not the semantics we need here.